botorch/acquisition/proximal.py

```diff
@@ -66,29 +68,63 @@ def __init__(
         self.register_buffer("proximal_weights", proximal_weights)
 
         # check model for train_inputs and single batch
-        if not hasattr(self.acq_func.model, "train_inputs"):
+        if not hasattr(model, "train_inputs"):
             raise UnsupportedError(
                 "Acquisition function model must have " "`train_inputs`."
             )
 
-        if (
-            self.acq_func.model.batch_shape != torch.Size([])
-            and self.acq_func.model.train_inputs[0].shape[1] != 1
-        ):
-            raise UnsupportedError(
-                "Proximal acquisition function requires a single batch model"
-            )
+        # change behavior depending on type of model
```

Comment:
Would you mind putting this into a helper function (e.g. `_validate_model`) so that the constructor doesn't get excessively long and becomes more readable? 

I also seems that by factoring out some of the checks into another function some of the code duplication could be avoided.